qt/python/mantidqt/project/encoderfactory.py

```diff
@@ -47,4 +48,9 @@ def register_encoder(cls, encoder, compatible_check=None):
         """
         if compatible_check is None:
             compatible_check = default_encoder_compatability_check
+        # check that a decoder of this class is not already registered
+        for existing_encoder in cls.encoder_list:
+            if encoder == existing_encoder[0]:
+                logger.notice("Encoder of this type already exists. Skipping")
+                return
```

Comment:
Could you help me understand the case where an encoder class can be registered twice? If this is the case I would be tempted to treat the factory more like a Python dict. If key exists then the value just overwrites that key without logging or atleast log at debug level. For a `dict` the dev is responsible for checking if something exists if they want to and this feels like a similar model?

Reply:
This related to an issue where the code to register the interfaces was called twice from MainWindow. Will refactor to use a dict